### DISCUSSION TITLE
How to insert all inlinks to other specific note with templater

### DISCUSSION THREAD

**Discussion by kristianberge** [SilentVoid13/Templater#930] [Help]
Hey,
Is there a way to use templater to insert all inlinks to lets say "note a" from another note?

 - Kristian

**Comment by Zachatoo** [comment]
To make sure we're using the same terminology, what does an "inlink" mean? Can you provide an example of what the state of the note should be before and after running this script?

**Reply by kristianberge** [reply]
Hi,
Thanks for following up. What I mean is, if there is for instance four documents, doc A, doc B, doc C and doc D. In both doc B and C there are links to doc A. I would like to be able to run a script from doc D which inserts all pages which are linking to doc A, in other word doc B and C.

The terminology "inlinks" I got from obsidian running the code dataview inline code: `$=dv.span(dv.current())` which gave me:
![CleanShot 2022-12-18 at 21 04 53@2x](https://user-images.githubusercontent.com/118932013/208317010-64986905-d72b-4aea-9412-53db4340f02c.png)

**Reply by Zachatoo** [reply]
So basically how the "Backlink in document" feature works for the [Backlinks](https://help.obsidian.md/Plugins/Backlinks) core plugin? From the Obsidian docs:

> If you want to see backlinks for multiple notes at the same time, you can enable Backlink in document under plugin options, to display backlinks in the same pane as your note.

Do you have any issues using that built in feature that works automatically as opposed to having to run a template to update the list?

**Reply by kristianberge** [reply]
well. sort of. but I want to insert hardcoded links into a note as part of a reference management system I'm building. When I am reading through and taking notes from medical research articles I often create links/pages for important quotes/statements. Many references/articles can have the same statements/link to the same page. When I write drafts for research papers I would like to option to insert the references of these statements, thus insert all the "inlinks" to a spesific statement without using the core plugin "Backlinks".

**Reply by AB1908** [reply]
Why don't you just write a query that views this data? Why insert it into the file? Once blacksmithgu/obsidian-dataview#42 is solved, that should take care of this use case as well.

**Reply by kristianberge** [reply]
AB1908, thank you for your reply. Yes, that is what I now am doing with the following code:

```
<%* 
var file = app.workspace.getActiveFile()
var t = await app.vault.read(file)
var s = t.replace("\[\[\[", "(`$= dv.page(\"").replace("\]\]\]", "\").file.inlinks`)")
app.vault.modify(file, s)
-%>
```

I was not aware of https://github.com/blacksmithgu/obsidian-dataview/issues/42 but this would solve my problem!
